components/brave_wallet/browser/zcash/zcash_shield_sync_service.cc

```diff
@@ -62,22 +63,17 @@ void ZCashShieldSyncService::OrchardBlockScannerProxy::ScanBlocks(
 }
 
 ZCashShieldSyncService::ZCashShieldSyncService(
-    ZCashRpc* zcash_rpc,
+    ZCashWalletService* zcash_wallet_service,
     const mojom::AccountIdPtr& account_id,
     const mojom::ZCashAccountShieldBirthdayPtr& account_birthday,
     const OrchardFullViewKey& fvk,
-    base::FilePath db_dir_path,
     base::WeakPtr<Observer> observer)
-    : zcash_rpc_(zcash_rpc),
+    : zcash_wallet_service_(zcash_wallet_service),
```

Comment:
why we need circular dependency here?
ZCashWalletService <-> ZCashShieldSyncService

Reply:
ZCashWalletService shares objects that are needed by zcash_shield_sync_service and subtasks like sync_state, zcash_rpc, keyring_service, etc.. So passing it here just simplifies accessing this members.